File: R/train/02.R
box::use(./lib/string[...])
box::use(./lib/functional[...])

words = list(
    list('contemplate', 9),
    list('conducive', 4),
    list('entrenched', 1),
    list('scrupulous', 5),
    list('ubiquitous', 7)
)

# index a list of pair(word, index)
ans_train_02 = words |>
    lapply(function(x) do.call(str.index, x)) |>
    str.concat() |>
    print()

File: hask/train/02.hs

words = [ ("contemplate", 9),
          ("conducive", 4),
          ("entrenched", 1),
          ("scrupulous", 5),
          ("ubiquitous", 7) ]

index :: [a] -> Int -> a
index list i = list !! i

ans_train_02 = [index (fst x) (snd x - 1) | x <- words]

